config/src/kura.rs

```diff
@@ -1,60 +1,33 @@
 //! Module for kura-related configuration and structs
-#![allow(clippy::std_instead_of_core)]
-use std::{num::NonZeroU64, path::Path};
+use core::num::NonZeroU64;
 
-use eyre::{eyre, Result};
-use iroha_config_base::derive::{Documented, Proxy};
+use eyre::Result;
+use iroha_config_base::{Configuration, Documented};
 use serde::{Deserialize, Serialize};
 
-const DEFAULT_BLOCKS_PER_STORAGE_FILE: u64 = 1000_u64;
-const DEFAULT_BLOCK_STORE_PATH: &str = "./storage";
-const DEFAULT_ACTOR_CHANNEL_CAPACITY: u32 = 100;
-
 /// `Kura` configuration.
-#[derive(Clone, Deserialize, Serialize, Debug, Documented, Proxy, PartialEq, Eq)]
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize, Serialize, Configuration, Documented)]
+#[serde(try_from = "ConfigurationBuilder")]
 #[serde(rename_all = "UPPERCASE")]
 #[config(env_prefix = "KURA_")]
 pub struct Configuration {
     /// Initialization mode: `strict` or `fast`.
-    pub init_mode: Mode,
+    #[config(default = "Mode::Strict")]
+    init_mode: Mode,
     /// Path to the existing block store folder or path to create new folder.
-    pub block_store_path: String,
+    #[config(default = "\"./storage\".to_owned()")]
+    block_store_path: String,
     /// Maximum number of blocks to write into a single storage file.
-    pub blocks_per_storage_file: NonZeroU64,
+    // SAFETY: Value is not zero
+    #[config(default = "unsafe { NonZeroU64::new_unchecked(1000_u64) }")]
```

Comment:
I would prefer `const panic!`. Also, for complex defaults you can define a separate const (AFAIU):

```rust
const DEFAULT_BLOCKS_PER_STORAGE: NonZeroU64 = match NonZeroU64::new(1000) {
  Some(x) => x,
  None => unreachable!()
};

// ...

#[config(default = "DEFAULT_BLOCKS_PER_STORAGE")]
```